### PR TITLE
Enable tab completion with fields

### PR DESCRIPTION
**Pull request by mtury**
This is an attempt at enabling tab completion with packet fields. It is not just a matter of quickly referencing long fields: I'd say it's even more useful as a way to remember/find out which fields you can get/set within an interpreter when you don't really know the packet you're dealing with.

I've been bothered by the lack of this feature time and again, so when I received a user request I tried to deal with it. The main issue was to reproduce the result of `dir()`, as Python 2 does not provide something like `super().__dir__()`.

Then we add the list of `default_fields`. I assumed that any value which might appear later on in `overloaded_fields` and `fields` should be present in `default_fields`. Does this seem right to you?

Also, I'm not sure how we could avoid the creation of lists with `keys()`. But I find it definitively good for UX.

Currently, `len(dir(TLSClientHello())) = 135` and `len(TLSClientHello().default_fields) = 13`
With this PR, counting `_superdir` and `__dir__`, you can check `len(dir(TLSClientHello())) = 150`

### CODE REVIEW

**Comment by gpotter2** [comment]
This sort of existed when using IPython, with
https://github.com/secdev/scapy/blob/22abf5e2f64a3fb7951fcf4dbe73da9fd6fe9b58/scapy/main.py#L489
If you add this, I guess you can remove that IPython only part (IPython also uses `__dir__`)

File: scapy/packet.py [hunk 1]
```diff
@@ -284,6 +284,25 @@ def __delattr__(self, attr):
...
+        """
+        Return a list of slots and methods, including those from subclasses.
+        """
+        attrs = set()
+        cls = self.__class__
+        if hasattr(cls, '__all_slots__'):
+            attrs.update(cls.__all_slots__)
+        for bcls in cls.__mro__:
+            if hasattr(bcls, '__dict__'):
+                attrs.update(bcls.__dict__.keys())
```
Comment (p-l-) [inline scapy/packet.py]: You don't need `.keys()`, plus with Python 2 this creates a useless list.

File: scapy/packet.py [hunk 2]
```diff
@@ -284,6 +284,25 @@ def __delattr__(self, attr):
...
+        return list(attrs)
```
Comment (p-l-) [inline scapy/packet.py]: This is not needed, `return attrs` should be enough.

File: scapy/packet.py [hunk 3]
```diff
@@ -284,6 +284,25 @@ def __delattr__(self, attr):
...
+
+    def __dir__(self):
+        """
+        Add fields to tab completion list.
+        """
+        return sorted(self._superdir() + list(self.default_fields))
```
Comment (p-l-) [inline scapy/packet.py]: You can simply write:
```Python
return sorted(itertools.chain(self._superdir(), self.default_fields))
```

**Comment by p-l-** [comment]
@gpotter2 I have the feeling it's better to have a generic Python solution rather than an iPython-specific one. What do you think?

**Comment by mtury** [comment]
I brought in the suggested changes in the first (revised) commit, and removed IPython-specific `complete_object` in the second one.

**Pull request merged**